### PR TITLE
Share to discourse

### PR DESCRIPTION
**Pull request by pelikhan**
Add optional "share to forum" button on share page. Probably need a better icon and text.

image
Used with https://github.com/Microsoft/pxt-arcade/pull/880

### CODE REVIEW

File: webapp/src/share.tsx [hunk 1]
```diff
@@ -493,36 +497,46 @@ class SocialButton extends data.Component<SocialButtonProps, {}> {
...
+                    twitterText = lf("Check out what I made with @{0}!", socialOptions.orgTwitterHandle);
+                }
+                url = `https://twitter.com/intent/tweet?url=${encodeURIComponent(shareUrl)}` +
+                    `&text=${encodeURIComponent(twitterText)}` +
+                    (socialOptions.hashtags ? `&hashtags=${encodeURIComponent(socialOptions.hashtags)}` : '') +
+                    (socialOptions.related ? `&related=${encodeURIComponent(socialOptions.related)}` : '');
+                break;
+            }
+            case "discourse": {
+                // https://meta.discourse.org/t/compose-a-new-pre-filled-topic-via-url/28074
```
Comment (mmoskal) [inline webapp/src/share.tsx]: maybe better to use the script title as the "title=" and put the URL in "body=" ?

Comment (mmoskal) [reply webapp/src/share.tsx]: and even if we don't have the title, I think the URL should be in the body

Comment (mmoskal) [reply webapp/src/share.tsx]: I think `comments` would be a good icon

**Comment by Jaqster** [comment]
How hard would it be to add a 'send by text' to a mobile phone number?  This is really how kids are communicating with each other - Text, Snapchat and Instagram.

File: webapp/src/share.tsx (hunk 1 above)
Comment (pelikhan) [reply webapp/src/share.tsx]: Discourse automatically extracts the title from a URL pasted in the title.

**Comment by abchatra** [comment]
Can the button be same size as facebook twitter?

**Comment by pelikhan** [comment]
You can design it in the UI meeting. If we make it an icon, I don’t think people will discover it since the icon is generic.

**Pull request merged**